### DISCUSSION TITLE
Import Rhino files with Discovery options

### DISCUSSION THREAD

**Discussion by dkowollik** [ansys/pyansys-geometry#1378] [Q&A]
Hi @RobPasMue,

I am struggling to get all the import options for Rhino files working that I usually use within Discovery or SpaceClaim. 
So far I am using: 
`from ansys.geometry.core import launch_modeler_with_geometry_service`
`modeler = launch_modeler_with_geometry_service()`
`model = modeler.open_file('my_rhino_file.3dm')`

In Discovery I used the following options (Sorry for the german version. Somehow Ansys 2024R2 installed Discovery in german):
![grafik](https://github.com/user-attachments/assets/17cf7fc6-8b23-4ea8-a46e-538285d4c47b)
![grafik](https://github.com/user-attachments/assets/21e177fc-960c-4f11-abc4-80d734352329)

Is it possible to use these options also in GeometryService?

**Comment by RobPasMue** [comment]
Hi @dkowollik! We do have ImportOptions available - see https://geometry.docs.pyansys.com/version/0.7/api/ansys/geometry/core/misc/options/ImportOptions.html#importoptions

The way to use them would be as follows:

```python
from ansys.geometry.core import launch_modeler_with_geometry_service
from ansys.geometry.core.misc import ImportOptions

# Launch the modeler
modeler = launch_modeler_with_geometry_service()

# Define your import options
import_options = ImportOptions(...)

# Open your file with given options
model = modeler.open_file("my_rhino_file.3dm", import_options=import_options)
```

Give it a try and let me know how it goes. If you are missing import options we can talk with the team and see if we can expose them, but probably not before the next release...

**Reply by dkowollik** [reply]
I already had a look at the ImportOptions dataclass before. The file type specific options are missing. In the traditional GUIs of Discovery and SpaceClaim I can define different Options for each file type that is supported by the traditional GUIs. 


```python
@dataclass
class ImportOptions:
    """Import options when opening a file.

    Parameters
    ----------
    cleanup_bodies : bool = False
        Simplify geometry and clean up topology.
    import_coordinate_systems : bool = False
        Import coordinate systems.
    import_curves : bool = False
        Import curves.
    import_hidden_components_and_geometry : bool = False
        Import hidden components and geometry.
    import_names : bool = False
        Import names of bodies and curves.
    import_planes : bool = False
        Import planes.
    import_points : bool = False
        Import points.
    """

    cleanup_bodies: bool = False
    import_coordinate_systems: bool = False
    import_curves: bool = False
    import_hidden_components_and_geometry: bool = False
    import_names: bool = False
    import_planes: bool = False
    import_points: bool = False
```

I need something like: 
```python
ImportOptions(
    optimize_imported_geometry = False, 
    check_geometry=False, 
    convert_rhino_layer_to_discovery_components=True, 
    import_facetted_data_as=Constants.VolumeAndSurfaceBodies, 
    stitch_surface=True)
```

**Reply by RobPasMue** [reply]
Gotcha... Let me bring it with the team as well and see what we can do. Again, this probably won't be available earlier than the next release.
